### PR TITLE
Add unit tests for MultiXMLConfigurationSerializer

### PR DESCRIPTION
**Pull request by Nirothipan**
## Automation tests
 - Unit tests 
   > 30%

### CODE REVIEW

File: modules/core/src/test/java/org/apache/synapse/config/xml/MultiXMLConfigurationSerializerTest.java [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+ */
+public class MultiXMLConfigurationSerializerTest {
+
+    private static final String SYNAPSE_XML = "synapse.xml";
+    private static final String KEY_DYNAMIC_SEQUENCE_1 = "dynamic_sequence_1";
+    private static final String KEY_DYNAMIC_ENDPOINT_1 = "dynamic_endpoint_1";
+    private static final String TEST_DIRECTORY_NAME = "serializeTestFolder";
+
+    private static final String DYNAMIC_ENDPOINT_1 =
+            "<endpoint xmlns=\"http://ws.apache.org/ns/synapse\">\n" +
```
Comment (sdkottegoda) [inline modules/core/src/test/java/org/apache/synapse/config/xml/MultiXMLConfigurationSerializerTest.java]: Line wrapping should happen before an operator

File: modules/core/src/test/java/org/apache/synapse/config/xml/MultiXMLConfigurationSerializerTest.java [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+    public void testSerialize2() throws IOException {
+        File file = new File(TEST_DIRECTORY_NAME);
+        if (!file.exists()) {
+            file.mkdir();
+            File xmlFile = new File(TEST_DIRECTORY_NAME + File.separator + "test.xml");
+            if (!xmlFile.exists()) {
+                xmlFile.createNewFile();
+            }
+        }
+        MultiXMLConfigurationSerializer serializer = new MultiXMLConfigurationSerializer(TEST_DIRECTORY_NAME);
```
Comment (sdkottegoda) [inline modules/core/src/test/java/org/apache/synapse/config/xml/MultiXMLConfigurationSerializerTest.java]: xmlFile.createNewFile() returns before the file is actually created in the file system. So it's better to check if the file exsists before proceeding further.

File: modules/core/src/test/java/org/apache/synapse/config/xml/MultiXMLConfigurationSerializerTest.java (hunk 1 above)
Comment (Nirothipan) [reply modules/core/src/test/java/org/apache/synapse/config/xml/MultiXMLConfigurationSerializerTest.java]: fixed this

File: modules/core/src/test/java/org/apache/synapse/config/xml/MultiXMLConfigurationSerializerTest.java (hunk 2 above)
Comment (Nirothipan) [reply modules/core/src/test/java/org/apache/synapse/config/xml/MultiXMLConfigurationSerializerTest.java]: fixed this

File: modules/core/src/test/java/org/apache/synapse/config/xml/MultiXMLConfigurationSerializerTest.java [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+        if (!file.exists()) {
+            file.mkdir();
+            if (!xmlFile.exists()) {
+                xmlFile.createNewFile();
+            }
+        }
+        Long startTime = System.currentTimeMillis();
+        do {
+            try {
+                Thread.sleep(1);
```
Comment (nuwanw) [inline modules/core/src/test/java/org/apache/synapse/config/xml/MultiXMLConfigurationSerializerTest.java]: can you increase the sleep to 500m

File: modules/core/src/test/java/org/apache/synapse/config/xml/MultiXMLConfigurationSerializerTest.java [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+            } catch (InterruptedException ignored) {
+            }
+        } while (!xmlFile.exists() && (System.currentTimeMillis() - startTime) < 5000);
+        if (xmlFile.exists()) {
+            MultiXMLConfigurationSerializer serializer = new MultiXMLConfigurationSerializer(TEST_DIRECTORY_NAME);
+            SynapseConfiguration configuration = new SynapseConfiguration();
+            serializer.serialize(configuration);
+            Assert.assertTrue("Error in serializing Synapse configuration.",
+                    new File(TEST_DIRECTORY_NAME + File.separator + SYNAPSE_XML).exists());
+        }
```
Comment (nuwanw) [inline modules/core/src/test/java/org/apache/synapse/config/xml/MultiXMLConfigurationSerializerTest.java]: test case should be failed if file fail to create

**Pull request merged**